app/services/notification-service.js

```diff
@@ -0,0 +1,86 @@
+/* global $ */
+import Ember from 'ember';
+
+export default Ember.Service.extend(Ember.Evented, {
+    authService: Ember.inject.service(),
+    topics: {},
+    topicsLoaded: false,
+    hasResponse: false,
+    hideRequest: false,
+    topicsUrl: '',
+    getTopicForResource: function(apiUri) {
+      let tps = this.get("topics");
+      return tps.get(apiUri);
+    },
+    urlHasTopic: function(apiUri) {
+      let tps = this.get("topics");
+      return tps.has(apiUri);
+    },
+    loadNotifications: function() {
+      this.set("hasResponse", false);
+      this.set("hideRequest", false);
+
+      let url = this.get('topicUrl');
+      if(!url) {
+        console.error('Unable to get available topics Url');
+        return;
+      }
+
+      let requestParams = {
+        method: 'get',
+        url: url,
+        timeout: 16000,
+        headers:{
+        }
+      };
+
+      requestParams.headers['content-type'] = 'application/json';
+      requestParams.headers['Authorization'] = this.get("authService").authHeader;
+      let that = this;
+
+      function handleResponse(xhResponse){
+        let responseData = xhResponse.responseJSON;
+        if(!responseData.entities) {
+          return;
+        }
+        let urlMap = new Map();
+        for(let i in responseData.entities) {
+          try {
+            let topic = responseData.entities[i];
+            if(!topic.publicApiTemplateUriPaths) {
+              continue;
+            }
+            for(let j = 0; j < topic.publicApiTemplateUriPaths.length; j++) {
+              let topicUrl = topic.publicApiTemplateUriPaths[j];
+              if (!topicUrl.includes('No related public API resource')) {
+                urlMap.set(topicUrl, topic.id);
+              }
+            }
+          } catch(err) {
+            console.error('Failed to load topics: ' + err);
+            return;
+          }
+        }
+        that.set("topics", urlMap);
+        that.set("topicsLoaded", (urlMap.size > 0));
+        console.log("loaded " + urlMap.size + " topics");
+      }
+      $.ajax(requestParams).then(function( data, textStatus, jqXHR  ) {
+        that.set("hasResponse", true);
+        that.set("hideRequest", true);
+        that.set("response", handleResponse(jqXHR));
+      }).catch(function(jqXHR){
+        console.log(jqXHR);
+        that.set("hasResponse", true);
+        that.set("hideRequest", true);
+        console.error("Error getting available topics: " + jqXHR.response.status + " " + jqXHR.responseText);
+      });
+    },
+    init() {
+      this._super(...arguments);
+      let searchSplit = window.location.search.split('//')[1];
+      let host = searchSplit.split('/')[0];
```

Comment:
Hi,

Not sure if it's my machine or it's the code, but when I tried to run this locally on my machine, the webpage doesn't load and I received an error on this line. According to the error message, the searchSplit variable is undefined, and the split function caused errors. Can you double-check if it works on your end? And let me know if I'm doing something wrong here. 

Thanks!

Reply:
I added error handling around getting the openApiUrl queryParameter,  You need to specify the openApiUrl parameter in your Url, normally this would always be called from a link that includes this. 

http://localhost:4200/?openApiUrl=https://api.inindca.com/api/v2/docs/swagger#token_type=bearer&access_token=<your auth token>&expires_in=172799